dashboard/app/controllers/activities_controller.rb

```diff
@@ -161,14 +161,13 @@ def track_progress_for_user
   end
 
   def track_progress_in_session
-    # hash of level_id => test_result
-    test_result = params[:testResult].to_i
-    old_result = client_state.level_progress(@level.id)
-
-    @new_level_completed = true if !Activity.passing?(old_result) && Activity.passing?(test_result)
-
     # track scripts
     if @script_level.try(:script).try(:id)
```

Comment:
This can happen when viewing a `/levels/[n]` link, but in those cases I don't think we should be tracking progress!